addons/fleet/wizard/fleet_vehicle_send_mail.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class FleetVehicleSendMail(models.TransientModel):
+    _name = 'fleet.vehicle.send.mail'
+    _inherit = 'mail.composer.mixin'
+    _description = 'Send mails to Drivers'
+
+    vehicle_ids = fields.Many2many('fleet.vehicle', string='Vehicles', required=True)
+    author_id = fields.Many2one('res.partner', 'Author', required=True, default=lambda self: self.env.user.partner_id.id)
+    template_id = fields.Many2one(domain=lambda self: [('model_id', '=', self.env['ir.model']._get('fleet.vehicle').id)])
+    attachment_ids = fields.Many2many(
+        'ir.attachment', 'fleet_vehicle_mail_compose_message_ir_attachments_rel',
+        'wizard_id', 'attachment_id', string='Attachments')
+
+    @api.onchange('template_id')
+    def _onchange_template_id(self):
+        self.attachment_ids = self.template_id.attachment_ids
+
+    def action_send(self):
+        self.ensure_one()
+        without_emails = self.vehicle_ids.driver_id.filtered(lambda a: not a.email)
+        if without_emails:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'danger',
+                    'message': _("The following vehicle drivers are missing an email address: %s.", ', '.join(without_emails.mapped(lambda a: a.name))),
```

Comment:
mapped accepts either a lambda function or a string with the name of the field you want to get. In this case, it's easier to just pass "name"
```suggestion
                    'message': _("The following vehicle drivers are missing an email address: %s.", ', '.join(without_emails.mapped("name"))),
```